lisa/operating_system.py

```diff
@@ -150,15 +220,41 @@ def type_name(cls) -> str:
     def name_pattern(cls) -> Pattern[str]:
         return re.compile(f"^{cls.type_name()}$")
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
+        raise NotImplementedError()
+
+    def _query_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> ExecutableResult:
         raise NotImplementedError()
 
     def _initialize_package_installation(self) -> None:
         # sub os can override it, but it's optional
         pass
 
+    def download_packages(
```

Comment:
It's similar with what `wget.get` can do. It doesn't need to be a method in os.

Reply:
> It's similar with what `wget.get` can do. It doesn't need to be a method in os.
yup.